plugins/CuraEngineBackend/StartSliceJob.py

```diff
@@ -222,32 +247,24 @@ def _buildExtruderMessage(self, stack):
         message = self._slice_message.addRepeatedMessage("extruders")
         message.id = int(stack.getMetaDataEntry("position"))
 
-        material_instance_container = stack.findContainer({"type": "material"})
+        settings = self._buildReplacementTokens(stack)
 
-        settings = {}
-        for key in stack.getAllKeys():
-            settings[key] = stack.getProperty(key, "value")
-            Job.yieldThread()
+        #Also send the material GUID. This is a setting in fdmprinter, but we have no interface for it.
+        material_instance_container = stack.findContainer({"type": "material"})
```

Comment:
I think this can just be `stack.material`, which will be faster, and check if it is `empty_material`.

Reply:
Indeed, it can. That removes the need for some error handling. I've made that change here: https://github.com/Ultimaker/Cura/pull/2375/commits/e55bb1e25c94b0d2e78a53372116fa60cda99971